azkaban-web-server/src/main/java/azkaban/imagemgmt/daos/ImageVersionDaoImpl.java

```
@@ -0,0 +1,132 @@
+
+  private static String INSERT_IMAGE_VERSION_QUERY =
+      "insert into image_versions ( path, description, version, type_id, state, "
+          + "release_tag, created_on, created_by) "
+          + "values (?, ?, ?, ?, ?, ?, ?, ?)";
+  private static String SELECT_IMAGE_VERSION_BASE_QUERY = "select iv.id, iv.path, iv.description, "
+      + "iv.version, it.type, iv.state, iv.release_tag, iv.created_on, iv.created_by, iv.modified_on, "
+      + "iv.modified_by from image_versions iv, image_types it where it.id = iv.type_id";
+
+  @Inject
+  public ImageVersionDaoImpl(DatabaseOperator databaseOperator, ImageTypeDao imageTypeDao) {
+    this.databaseOperator = databaseOperator;
+    this.imageTypeDao = imageTypeDao;
+  }
+
+  @Override
+  public int createImageVersion(ImageVersion imageVersion) {
+    ImageType imageType = imageTypeDao.getImageTypeByType(imageVersion.getType())
+        .orElseThrow(() -> new RuntimeException("Unable to fetch image type metadata. Invalid "
+            + "image type : "+imageVersion.getType()));
+    final SQLTransaction<Long> insertAndGetSpaceId = transOperator -> {
+      String currentTime = DateTime.now().toLocalDateTime().toString();
+      transOperator.update(INSERT_IMAGE_VERSION_QUERY, imageVersion.getPath(), imageVersion.getDescription(),
+          imageVersion.getVersion(), imageType.getId(), imageVersion.getState().getStateValue(),
+          imageVersion.getReleaseTag(), currentTime, imageVersion.getCreatedBy());
+      transOperator.getConnection().commit();
+      return transOperator.getLastInsertId();
+    };
+
+    int imageVersionId = 0;
+    try {
+      /* what will happen if there is a partial failure in
+         any of the below statements?
+         Ideally all should happen in a transaction */
+      imageVersionId = databaseOperator.transaction(insertAndGetSpaceId).intValue();
+    } catch (SQLException e) {
+      log.error("Unable to create the image version metadata", e);
```

Can you please throw an exception here?

---

Done.